src/api/spec/controllers/webui/patchinfo_controller_spec.rb

```diff
@@ -0,0 +1,269 @@
+require 'rails_helper'
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the Patchinfo methods
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::PatchinfoController, vcr: true do
+  let(:user) { create(:user, login: 'macario') }
+  let(:other_user) { create(:confirmed_user, login: 'gilberto')}
+  let(:other_package) { create(:package, project: user.home_project, name: 'other_package') }
+  let(:patchinfo_package) do
+    patchinfo = Patchinfo.new.create_patchinfo(user.home_project_name, nil)
```

Comment:
Patchinfo is an ActiveXml model... I don't know if this is better or not, have you tried to do a factory of any those non ActiveRecord models?